rllib/agents/trainer.py

```diff
@@ -613,7 +613,7 @@ def get_scope():
     def _stop(self):
         if hasattr(self, "workers"):
             self.workers.stop()
-        if hasattr(self, "optimizer"):
+        if hasattr(self, "optimizer") and self.optimizer:
```

Comment:
I was wondering about these things. If Trainer is not sure to have a prop "optimizer" or not, should this prop not be part of the Trainer class (and then sometimes be None)? I know we do this in many places, so not urgent, but we should think about fixing these.

Reply:
Probably the right solution is to make it a required prop of a subclass of Trainer.

I'll leave it alone for now, since we can probably remove .optimizer entirely once the pipeline implementation becomes the default.